koap/src/commonMain/kotlin/Message.kt

```diff
@@ -408,7 +410,7 @@ sealed class Message {
                     payload.contentEquals(other.payload))
 
         override fun hashCode(): Int =
-            Objects.hash(code, token, options, payload.contentHashCode())
+                ObjectHasherFactory.createObjectHasher().hash(code, token!!, options, payload.contentHashCode()).toInt()
```

Comment:
We shouldn't have `!!` on the `token`, instead the `hash` function should omit that field from the hash calculation.

Reply:
Fixed